Source/Evolve/WarpXEvolve.cpp

```diff
@@ -325,19 +330,17 @@ WarpX::Evolve (int numsteps)
         }
         multi_diags->FilterComputePackFlush( step );
 
-        if (cur_time >= stop_time - 1.e-3*dt[0]) {
-            break;
-        }
-
-        if (warpx_py_afterstep) warpx_py_afterstep();
-
         // inputs: unused parameters (e.g. typos) check after step 1 has finished
         if (!early_params_checked) {
             amrex::Print() << "\n"; // better: conditional \n based on return value
             amrex::ParmParse().QueryUnusedInputs();
             early_params_checked = true;
         }
 
+        if (cur_time >= stop_time - 1.e-3*dt[0]) {
```

Comment:
@WeiqunZhang I saw that you changed a bit on this line in 422ee81125
Do you know the logic for the dt epsilon here with respect to @peterscherpelz's question?

> (By the way, I assume the reason that break is there is so that if someone intends to run a single step with cur_time minutely smaller than stop_time it'll still run one step? Otherwise it seems like that final break should just be in the initial for loop statement.)

Reply:
I don't think I can remember the detail.